packages/material-ui/src/FieldTemplate/WrapIfAdditional.tsx

```diff
@@ -0,0 +1,92 @@
+import React from "react";
+
+import { utils } from "@rjsf/core";
+import { JSONSchema7 } from "json-schema";
+
+import Grid from "@material-ui/core/Grid";
+import FormControl from "@material-ui/core/FormControl";
+import Input from "@material-ui/core/Input";
+import InputLabel from "@material-ui/core/InputLabel";
+
+import IconButton from "../IconButton/IconButton";
+
+const { ADDITIONAL_PROPERTY_FLAG } = utils;
+
+type WrapIfAdditionalProps = {
+  children: React.ReactElement;
+  classNames: string;
+  disabled: boolean;
+  id: string;
+  label: string;
+  onDropPropertyClick: (index: string) => (event?: any) => void;
+  onKeyChange: (index: string) => (event?: any) => void;
+  readonly: boolean;
+  required: boolean;
+  schema: JSONSchema7;
+};
+
+const WrapIfAdditional = ({
+  children,
+  classNames,
+  disabled,
+  id,
+  label,
+  onDropPropertyClick,
+  onKeyChange,
+  readonly,
+  required,
+  schema,
+}: WrapIfAdditionalProps) => {
+  const readonlyAsDisabled = true;
+  const keyLabel = `${label} Key`; // i18n ?
+  const additional = schema.hasOwnProperty(ADDITIONAL_PROPERTY_FLAG);
+  const btnStyle = {
+    flex: 1,
+    paddingLeft: 6,
+    paddingRight: 6,
+    fontWeight: "bold",
+  };
+
+  if (!additional) {
+    return <div className={classNames}>{children}</div>;
+  }
+
+  const handleBlur = ({ target }: React.FocusEvent<HTMLInputElement>) =>
+    onKeyChange(target.value);
+
+  return (
+    <Grid container={true} key={`${id}-key`} alignItems="center">
```

Comment:
Adding some spacing between the key/value fields will make it look nicer:
```suggestion
    <Grid container={true} key={`${id}-key`} alignItems="center" spacing={2}>
```